src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalCpxSvDiscoverFromLongReadsSpark.java

```diff
@@ -0,0 +1,228 @@
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.BetaFeature;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.StructuralVariationSparkProgramGroup;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignedContig;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignmentInterval;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.FileUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.EnumMap;
+import java.util.HashSet;
+import java.util.Set;
+
+
+/**
+ * This tool takes a SAM file containing alignments of single-ended long read
+ * (be it long read sequencing, or contigs assembled from standard Illumina short reads),
+ * searches for reads with complicated split alignments indicating the presence of complex structural variations (cxSV),
+ * and outputs a custom file format containing how affected reference segments are rearranged on the provided sample
+ * where the long reads are from.
+ */
+@CommandLineProgramProperties(summary="Parses a SAM file containing long reads alignments, and outputs cxSV rearrangements.",
+        oneLineSummary="Parses a long read SAM file, and outputs cxSV rearrangements.",
+        usageExample = "InternalCpxSvDiscoverFromLongReadsSpark \\" +
+                "-I /path/to/my/dir/longReads.sam \\" +
+                "-O /path/to/my/dir/outputDir \\" +
+                "-R /path/to/my/reference/reference.2bit --fastaReference /path/to/my/reference/reference.fasta",
+        programGroup = StructuralVariationSparkProgramGroup.class)
+@BetaFeature
+public final class InternalCpxSvDiscoverFromLongReadsSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+    private final Logger localLogger = LogManager.getLogger(InternalCpxSvDiscoverFromLongReadsSpark.class);
+
+    @ArgumentCollection
+    private StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection
+            discoverStageArgs
+            = new StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection();
+
+    @Argument(doc = "file containing non-canonical contig names (e.g chrUn_KI270588v1) in the reference, human reference assumed when omitted",
```

Comment:
Since in the default it's assuming that chromosome names start with "chr", the default is more accurately "human hg19 or hg38".